pkg/plugin/cmd.go

```diff
@@ -394,3 +427,70 @@ func copyAndStripPod(pod *corev1.Pod, targetContainer string) *corev1.Pod {
 
 	return copied
 }
+
+// createNodePortServiceAndEndpoints  use nodeport type service to connect the speficy agent pod
+func (o *DebugOptions) createNodePortServiceAndEndpoints(ns, name, agentIP string, agentPort int) (string, int32, error) {
```

Comment:
As for #13 , if we cannot access the host ip, neither can we access the node port. Maybe we should try port-forward as @random-ua suggested

Reply:
yeah, this not a security  method, but if we have access to create service(it will delete the service automatic), that mean we can have access the node port. the port-forward method is more  complicated than this, we can use this method  temporary and remove it after the other method is merged.

Reply:
the insecurity phase is that  when we use the node port , the other people also can access the node port.

i think this sentence  `if we cannot access the host ip, neither can we access the node port.` is 
not all right, the host ip and cluster ip is not in same network segment and don't config the route also can lead to cannot access the host ip. but if we has the `.kube/config` file, we shou have the access to visit the  host ip.

Reply:
Actually, debug-agent runs in `hostNetwork` mode and the cli side connects to `hostIP`. So a `NodePort` service will not provide extra accessibility. 
The accessing problem typically arise in public cloud like GKE. In such environment, kubelet nodes typically don't have public IP, thus the `hostIP` is not accessible.
Moreover, `kubectl-debug` vendors the kubectl package, so it is as straightforward as creating a service. I can provide some basic ideas and code links if you are interested in implementing this😄

Reply:
yeah, i'm glad to try it.

Reply:
i'm sorry for that i have some mistake about nodeport

Reply:
It's okay, no need to be sorry bro. Both of us has things to learn and your enthusiasm for open-source is definitely awesome!

Reply:
and I will have spare time to write a rough port-foward design document tonight.